Windows build fails: TokioIo not imported for windows_named_pipe code

Hola, 

I just got a notification that one of my github action crashed. 
So I investigated quickly and my agent came up with the following (see below). I use tauri to build a windows application, as a refrence. It made sense to me what he figured out. So i drop here an issue.

thanks for the work!

Best regards,
Frank

---

## Version
0.12.27

## Platform
Windows (tested on Windows Server 2022 via GitHub Actions)

## Description
Windows builds fail when reqwest is compiled without TLS or SOCKS features. The `TokioIo` type is not imported on Windows, but is used by the new Windows Named Pipe code added in PR #2789.

## Error
```
error[E0433]: failed to resolve: use of undeclared type `TokioIo`
   --> src/connect.rs:726:17
```

## Cause
The `TokioIo` import has a cfg condition that doesn't include Windows:

```rust
#[cfg(any(feature = "socks", feature = "__tls", unix))]
use hyper_util::rt::TokioIo;
```

But the Windows Named Pipe code at line ~720 uses `TokioIo::new()` unconditionally on Windows:

```rust
#[cfg(target_os = "windows")]
let svc = {
    // ...
    async move { ClientOptions::new().open(pipe).map(TokioIo::new) }
};
```

## Suggested Fix
Add `target_os = "windows"` to the import cfg:

```rust
#[cfg(any(feature = "socks", feature = "__tls", unix, target_os = "windows"))]
use hyper_util::rt::TokioIo;
```

## Introduced In
PR #2789

--- a/src/connect.rs
+++ b/src/connect.rs
@@ -5,7 +5,7 @@ use http::uri::Scheme;
 use http::Uri;
 use hyper::rt::{Read, ReadBufCursor, Write};
 use hyper_util::client::legacy::connect::{Connected, Connection};
-#[cfg(any(feature = "socks", feature = "__tls", unix))]
+#[cfg(any(feature = "socks", feature = "__tls", unix, target_os = "windows"))]
 use hyper_util::rt::TokioIo;
 #[cfg(feature = "default-tls")]
 use native_tls_crate::{TlsConnector, TlsConnectorBuilder};

